libretro-rs/src/core_macro.rs

```diff
@@ -131,17 +130,17 @@ macro_rules! libretro_core {
       }
 
       #[no_mangle]
-      extern "C" fn retro_get_region() -> libc::c_uint {
+      extern "C" fn retro_get_region() -> c_uint {
         instance_ref(|instance| instance.on_get_region())
       }
 
       #[no_mangle]
-      extern "C" fn retro_get_memory_data(id: libc::c_uint) -> *mut () {
+      extern "C" fn retro_get_memory_data(id: c_uint) -> *mut () {
         instance_mut(|instance| instance.on_get_memory_data(id))
       }
 
       #[no_mangle]
-      extern "C" fn retro_get_memory_size(id: libc::c_uint) -> libc::size_t {
+      extern "C" fn retro_get_memory_size(id: c_uint) -> usize {
```

Comment:
In a cursory search, there seems to be a lot of uncertainty around `libc::size_t` being equivalent to `usize`, even within official rust code and documents. I think we should leave this as `libc::size_t` (or the `ffi` equivalent) until this is resolved.

Reply:
From following the discussions on [Strict Provenance](https://github.com/rust-lang/rust/issues/95228), Rust conflating `size_t` and `uintptr_t` wasn't ideal, but the distinction seems to only matter in *really* exotic architectures. The only one I know of is [CHERI](https://www.cl.cam.ac.uk/research/security/ctsrd/cheri/), which is specifically mentioned in the Strict Provenance discussion. It's a research architecture that has a 64-bit `size_t` but 128-bit `intptr_t` since it stores metadata in its pointers to track the regions of memory the pointer is allowed to access. The Strict Provenance tracking issue doesn't mention any other architectures like this. And `usize` certainly shouldn't be smaller than `size_t` since that'd mean there's objects larger than the address space.

Case in point, Bindgen is currently using `usize` to represent `size_t` in `libretro.h`; as long as Bindgen is doing that, there's no point in using `libc::size_t` elsewhere, since the create will only compile as long as those two types are aliases of each other. And a peek at [the libc crate's repo on Github](https://github.com/rust-lang/libc/search?q=%22size_t+%3D%22) shows that `size_t` is consistently aliased to `usize` (either directly, or transitively through `uintptr_t`.)

One last data point is the experimental [core::ffi::c_size_t](https://doc.rust-lang.org/core/ffi/type.c_size_t.html):
> Equivalent to C’s size_t type, from stddef.h (or cstddef for C++).
> 
> This type is currently always [usize](https://doc.rust-lang.org/core/primitive.usize.html), however in the future there may be platforms where this is not the case.